lab_1/main.py

```diff
@@ -1,94 +1,121 @@
-"""
-Lab 1
-A concordance extraction
-"""
+import math
 
 
+def checker(con_object, type_ob):
+    if type_ob is list and type(con_object) is list:
+        len_ = len(con_object) > 0
+        bull_ = None not in con_object and bool not in con_object
+
+        if len_ and bull_:
+            return True
+    elif type_ob is str and type(con_object) is str and len(con_object) >= 0:
+            return True
+    elif type_ob is int and type(con_object) is int and con_object >= 0:
+        return True
+
+    return False
+
+
+# токенезирование текста - пунктуация и заглавные
 def tokenize(text: str) -> list:
-    """
-    Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
-    :param text: the initial text
-    :return: a list of lowercased tokens without punctuation
-    e.g. text = 'The weather is sunny, the man is happy.'
-    --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
-    """
-    pass
+    tokens = []
 
+    for token in text.lower().split():
+        token = "".join([a for a in token if a.isalpha()])
+        tokens.append(token)
+    return tokens
 
+
+# получение стоп-слов
+def get_stop_words() -> list:
+    with open("stop_words.txt") as file:
+        stop_words = file.readlines()
+    return stop_words
+
+
+# удаление стоп-слов
 def remove_stop_words(tokens: list, stop_words: list) -> list:
-    """
-    Removes stop words
-    :param tokens: a list of tokens
-    :param stop_words: a list of stop words
-    :return: a list of tokens without stop words
-    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
-    stop_words = ['the', 'is']
-    --> ['weather', 'sunny', 'man', 'happy']
-    """
-    pass
+    if checker(tokens, list):
+        tokens_clean = [token for token in tokens if token not in stop_words]
+        return tokens_clean
+    else:
+        return []
 
 
+# подсчет частотности
 def calculate_frequencies(tokens: list) -> dict:
-    """
-    Calculates frequencies of given tokens
-    :param tokens: a list of tokens without stop words
-    :return: a dictionary with frequencies
-    e.g. tokens = ['weather', 'sunny', 'man', 'happy']
-    --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
-    """
-    pass
+    if checker(tokens, list):
+        frequencies = {token: tokens.count(token) for token in tokens if type(token) is str}
```

Comment:
You have `checker` function for type checking, right?

Reply:
changed, thanks